pkg/connector/backfill.go

```diff
@@ -19,11 +20,28 @@ import (
 
 // getTakeoutID blocks until the takeout ID is available.
 func (t *TelegramClient) getTakeoutID(ctx context.Context) (takeoutID int64, err error) {
-	log := zerolog.Ctx(ctx).With().Str("function", "getTakeoutID").Logger()
+	// Always stop the takeout timeout timer
+	if t.stopTakeoutTimer != nil {
+		t.stopTakeoutTimer.Stop()
+	}
+
+	// After we get the takeout ID, reset the timer.
+	defer func() {
+		t.stopTakeoutTimer = time.AfterFunc(max(time.Hour, time.Duration(t.main.Bridge.Config.Backfill.Queue.BatchDelay*2)), sync.OnceFunc(func() { t.stopTakeout(ctx) }))
+	}()
+
 	if t.userLogin.Metadata.(*UserLoginMetadata).TakeoutID != 0 {
+		// Resume fetchnig dialogs using takeout and enqueueing them for
+		// backfill.
+		go func() {
+			if err = t.takeoutDialogs(ctx, takeoutID); err != nil {
+				log.Err(err).Msg("Failed to takeout dialogs")
+			}
+		}()
```

Comment:
Doesn't this call the function on every FetchMessages call now? also doesn't make much difference because the lock and check inside, just a bunch of unnecessary goroutines

Reply:
I mean there will probably be a lot of FetchMessages calls while the bridge is running, so all of those are going to call this even if it was already called